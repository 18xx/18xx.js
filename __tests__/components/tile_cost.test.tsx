import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import TileCost from '../../src/components/tile_cost';

import Point from '../../src/point';

describe('TileCost', () => {
  describe('#toString', () => {
    describe('when shape is a triangle', () => {
      it('returns an svg representation of a triangle', () => {
        const subject: ReactElement<TileCost> = (
          <TileCost amount={10} color='#ccc' shape='triangle' />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when shape is not a triangle', () => {
      it('returns an svg representation of a square', () => {
        const subject: ReactElement<TileCost> = (
          <TileCost amount={14} color='#ddd' shape='square' />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });

    describe('when locations is set', () => {
      it('returns an svg located at the point specified', () => {
        const subject: ReactElement<TileCost> = (
          <TileCost
            amount={20}
            color='white'
            location={new Point(3, 7)}
            shape='square' />
        );
        expect(renderer.create(subject)).toMatchSnapshot();
      });
    });
  });
});
