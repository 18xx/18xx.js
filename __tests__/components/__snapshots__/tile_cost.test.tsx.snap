exports[`TileCost #toString when locations is set returns an svg located at the point specified 1`] = `
<g>
  <rect
    fill="white"
    height={20}
    stroke="#333"
    width={20}
    x={3}
    y={7} />
  <text
    textAnchor="middle"
    x={13}
    y={23}>
    20
  </text>
</g>
`;

exports[`TileCost #toString when shape is a triangle returns an svg representation of a triangle 1`] = `
<g>
  <polygon
    fill="#ccc"
    points="55.42562584220407,9 75.42562584220407,39 35.42562584220407,39"
    stroke="#333" />
  <text
    textAnchor="middle"
    x={55.42562584220407}
    y={35}>
    10
  </text>
</g>
`;

exports[`TileCost #toString when shape is not a triangle returns an svg representation of a square 1`] = `
<g>
  <rect
    fill="#ddd"
    height={20}
    stroke="#333"
    width={20}
    x={45.42562584220407}
    y={14} />
  <text
    textAnchor="middle"
    x={55.42562584220407}
    y={30}>
    14
  </text>
</g>
`;
